system_meta/models.py

```diff
@@ -77,10 +79,37 @@ class Product(SafeDeleteModel, SoftDeleteActiveModel, TimestampedModel):
         null=True,
         help_text="System-specific data for the product (in JSON).",
     )
+    image_metadata = models.JSONField(
+        blank=True,
+        null=True,
+        help_text="Image metadata including URL, alt text, and description (in JSON).",
+    )
 
     objects = SafeDeleteManager()
     all_objects = models.Manager()
 
+    def save(self, *args, **kwargs):
+        # Retrieve image data from the API
+        try:
```

Comment:
We really shouldn't be blocking the save operation by spawning a request that may or may not work. This should move into a task instead, I think, that gets queued here. If the Learn API fails (or is slow for whatever reason) in a task, it won't block the save op and we can then also requeue it to try again later if we need to. (Also you won't have to mock request, just the task function.)